src/zenml/integrations/github/secrets_managers/github_secrets_manager.py

```diff
@@ -0,0 +1,361 @@
+        prefix. Otherwise, this will return all GitHub **Repository** secrets
+        created by ZenML.
+
+        Args:
+            include_prefix: Whether or not the internal prefix that is used to
+                differentiate ZenML secrets from other GitHUb secrets should be
+                included in the returned names.
+        """
+        if inside_github_action_environment():
+            potential_secret_keys = list(os.environ)
+        else:
+            logger.info(
+                "Fetching list of secrets for repository %s/%s",
+                self.owner,
+                self.repository,
+            )
+            response = self._send_request("GET", params={"per_page": 100})
+            potential_secret_keys = [
+                secret_dict["name"]
+                for secret_dict in response.json()["secrets"]
+            ]
+
+        keys = [
+            _convert_secret_name(key, remove_prefix=not include_prefix)
+            for key in potential_secret_keys
+            if key.startswith(GITHUB_SECRET_PREFIX)
+        ]
+
+        return keys
+
+    def register_secret(self, secret: BaseSecretSchema) -> None:
+        """Registers a new secret.
+
+        Args:
+            secret: The secret to register.
+        """
+        if self._has_secret(secret.name):
+            raise KeyError(
+                f"A secret with name '{secret.name}' already exists for this "
+                "GitHub repository. If you want to register a new value for "
+                f"this secret, please run `zenml secret delete {secret.name}` "
+                f"followed by `zenml secret register {secret.name} ...`."
+            )
+
+        secret_dict = {
+            SECRET_SCHEMA_DICT_KEY: secret.TYPE,
+            SECRET_CONTENT_DICT_KEY: secret.content,
+        }
+        secret_value = string_utils.b64_encode(json.dumps(secret_dict))
+        encrypted_secret, public_key_id = self._encrypt_secret(
+            secret_value=secret_value
+        )
+        body = {
+            "encrypted_value": encrypted_secret,
+            "key_id": public_key_id,
+        }
+
+        full_secret_name = _convert_secret_name(secret.name, add_prefix=True)
+        self._send_request("PUT", resource=f"/{full_secret_name}", json=body)
+
+    def update_secret(self, secret: BaseSecretSchema) -> NoReturn:
+        """Update an existing secret.
+
+        Args:
+            secret: The secret to update.
+
+        Raises:
+            NotImplementedError: Always, as this functionality is not possible
+                using GitHub secrets which doesn't allow us to retrieve the
+                secret values outside of a GitHub Actions environment.
+        """
+        raise NotImplementedError(
+            "Updating secrets is not possible with the GitHub secrets manager "
+            "as it is not possible to retrieve GitHub secrets values outside "
+            "of a GitHub Actions environment."
+        )
```

Comment:
Does updating really require to fetch a secret? Can't we just let the user create or update keys in the secret no matter if it exists or not?

Reply:
As the secret is stored as an encoded dictionary (and this encoded value I can't fetch), it's not possible to update/create/delete keys from this secret at any point.

Reply:
Alright, makes sense